portals/publisher/src/main/webapp/source/src/app/components/Apis/Details/Environments/Environments.jsx

```diff
@@ -1483,6 +1511,246 @@ export default function Environments() {
+                    }}
+                    name={row.name}
+                    onChange={handleSelect}
+                    margin='dense'
+                    variant='outlined'
+                    style={{ width: '50%' }}
+                    disabled={api.isRevision || !filteredRevisions || filteredRevisions.length === 0}
+                >
+                    {filteredRevisions && filteredRevisions.length !== 0 && filteredRevisions.map((number) => (
+                        <MenuItem value={number.id}>{number.displayName}</MenuItem>
+                    ))}
+                </TextField>
+                <Button
+                    className={classes.button2}
+                    disabled={
+                        api.isRevision ||
+                        !selectedRevision.some((r) => r.env === row.name && r.revision) ||
+                        !selectedVhosts.some((v) => v.env === row.name && v.vhost) ||
+                        (api.advertiseInfo && api.advertiseInfo.advertised) ||
+                        isDeployButtonDisabled
+                    }
+                    variant='outlined'
+                    onClick={() =>
+                        deployRevision(
+                            selectedRevision.find((r) => r.env === row.name).revision,
+                            row.name,
+                            selectedVhosts.find((v) => v.env === row.name).vhost,
+                            selectedRevision.find((r) => r.env === row.name).displayOnDevPortal
+                        )
+                    }
+                >
+                    <FormattedMessage
+                        id='Apis.Details.Environments.Environments.deploy.button'
+                        defaultMessage='Deploy'
+                    />
+                </Button>
+            </div>
+        );
+    }
+
+    /**
+     * Get the deployment component based on the environment and revision status.
+     * @param {*} row Row
+     * @param {*} allEnvRevisionMapping All environment revision mapping
+     * @returns {JSX.Element} The JSX element representing the deployment component
+     */
+    function getDeployedRevisionComponent(row, allEnvRevisionMapping) {
+        const deployingGateway = allEnvRevisionMapping.find(gateway => {
+            return gateway.name === row.name});
+        const gatewayRevisions = deployingGateway?.revisions;
+    
+        if (!gatewayRevisions.length) {
+            // Content to display when there is no revision
+            return (
+                <FormattedMessage
+                    id='Apis.Details.Environments.Environments.status.not.deployed'
+                    defaultMessage='No Revision Deployed'
+                />
+            );
+            
+        }
+
+        const approvedDeployment =gatewayRevisions.find(revision => {
+            return revision.deploymentInfo.some(info => (info.status === null || info.status === "APPROVED") && 
+            info.name === row.name);});
```

Comment:
Let's fix the formatting here for readability.

Reply:
fixed with https://github.com/wso2/apim-apps/commit/feb7a470cc8167eb79976df580ae001188e69a10